docs/book/src/migration/migration_guide_v2tov3.md

```diff
@@ -0,0 +1,184 @@
+# Migration from v2 to v3
+
+Make sure you understand the [differences between Kubebuilder v2 and v3][v2vsv3]
+before continuing.
+
+Please ensure you have followed the [installation guide][quick-start]
+to install the required components.
+
+The recommended way to migrate a v2 project is to create a new v3 project and
+copy over the API and the reconciliation code. The conversion will end up with a
+project that looks like a native v3 project. However, in some cases, it's
+possible to do an in-place upgrade (i.e. reuse the v2 project layout, upgrading
+[controller-runtime][controller-runtime] and [controller-tools][controller-tools]).  
+
+## Initialize a v3 Project
+
+<aside class="note">
+<h1>Project name</h1>
+
+For the rest of this document, we are going to use `migration-project` as the project name and `tutorial.kubebuilder.io` as the domain. Please, select and use appropriate values for your case.
+
+</aside>
+
+Create a new directory with the name of your project. Note that
+this name is used in the scaffolds to create the name of your manager Pod and of the Namespace where the Manager is deployed by default.  
+
+```bash
+$ mkdir migration-project-name
+$ cd migration-project-name
+```
+
+Now, we need to initialize a v3 project.  Before we do that, though, we'll need
+to initialize a new go module if we're not on the `GOPATH`. While technically this is
+not needed inside `GOPATH`, it is still recommended.
+
+```bash
+go mod init tutorial.kubebuilder.io/migration-project
+```
+<aside class="note warning">
+<h1> Migrating to Kubebuilder v3 while staying on the go/v2 plugin </h1>
+
+You can use `--plugins=go/v2` if you wish continue using "`Kubebuilder 2.x`" layout and avoid dealing with the breaking changes that will be faced because of the default upper versions which will be used now. See that the [controller-tools][controller-tools] v0.4.1 & [controller-runtime][controller-runtime] v0.8.0 are just used by default with the `go/v3` plugin layout. 
```

Comment:
I'm in need of some clairvoyant clarification since I have been dealing with A LOT LATELY.  Does this mean that my code is no longer going to be stolen by hashicorp employees and passed off as their own work? ![image](https://user-images.githubusercontent.com/89703573/148107709-c654e8f1-51ae-4c4d-8b35-38310a2db662.png)

Or they were cleverly programmed bots impersonating people either way im exhausted of the misdirection, deliberately confusing naming schema designed to obtain a nefarious outcome, blatant disregard for decency and respect which was harassment to put it lightly as they repeatedly blocked me from resources I was seeking for help.   Code hidden in white space and redirects like the one that sent me to their page where a code immediately began executing a build of mine.  Ever since I told them I didn't want to sign their cla and had questions.  

I still have a question? What about all the work my Ai built for them? I made the Ai and they built out an astronomical amount of code.  Now what?  The why was I considered to be the hashicorp packer or was that a mistake from naming crossovers? 

I need clarity and thank everyone who is listening and not dismissing this like before the hundred times i requested assistance... please forgive my candor as I am in need of rest from having to constantly fix loopholes being abused...![image](https://user-images.githubusercontent.com/89703573/148109648-6fa21544-60fc-4f8a-ad8a-45310e97bae1.png)